crates/markdown_preview/src/markdown_elements.rs

```diff
@@ -18,21 +18,18 @@ pub enum ParsedMarkdownElement {
 }
 
 impl ParsedMarkdownElement {
-    pub fn source_range(&self) -> Range<usize> {
+    pub fn source_range(&self) -> Option<Range<usize>> {
         match self {
-            Self::Heading(heading) => heading.source_range.clone(),
-            Self::ListItem(list_item) => list_item.source_range.clone(),
-            Self::Table(table) => table.source_range.clone(),
-            Self::BlockQuote(block_quote) => block_quote.source_range.clone(),
-            Self::CodeBlock(code_block) => code_block.source_range.clone(),
-            Self::Paragraph(text) => match &text[0] {
-                MarkdownParagraphChunk::Text(t) => t.source_range.clone(),
-                MarkdownParagraphChunk::Image(image) => match image {
-                    Image::Web { source_range, .. } => source_range.clone(),
-                    Image::Path { source_range, .. } => source_range.clone(),
-                },
+            Self::Heading(heading) => Some(heading.source_range.clone()),
```

Comment:
Can pull one `Some(` out to wrap the whole match.

Reply:
Good point!